docs/docsite/rst/dev_guide/developing_plugins.rst

```diff
@@ -36,6 +36,17 @@ You should return errors encountered during plugin execution by raising ``Ansibl
     except Exception as e:
         raise AnsibleError('Something happened, this was original exception: %s' % to_native(e))
 
+Since Ansible evaluates variables lazily, filter and test plugins should propagate the exceptions ``jinja2.exceptions.UndefinedError`` and ``AnsibleUndefinedVariable``.
```

Comment:
is 'lazily' a.. erm... technical term? :-)  Or alternately, would most developers know what you mean by 'Ansible evaluates variables lazily...'

Reply:
@samccann It _is_ a technical term (meaning it only evaluates them when needed -- if it's never used, it's never evaluated).

I think most with programming experience would know what it means. But not sure if the average sysadmin would or not.

Reply:
"lazily" could just as well be replaced with "only when they are needed" or "only when they are used"